collectors/framework/framework.go

```diff
@@ -220,6 +222,32 @@ func (a *AvroDatum) transform(nodeInfo collectors.NodeInfo) (producers.MetricsMe
 		return pmm, err
 	}
 
+	decodedDps, ok := dps.([]interface{})
+	if !ok {
+		return pmm, fmt.Errorf("Could not decode datapoints %v", dps)
+	}
+
+	const metricsValueKey = "dcos.metrics.value"
+	for _, dp := range decodedDps {
+		dpr, ok := dp.(*goavro.Record)
+		if !ok {
+			fwColLog.Debugf("Bad datapoint record encountered: %v", dp)
+			continue
+		}
+		value, err := dpr.Get(metricsValueKey)
+		if err != nil {
+			fwColLog.Debugf("Datum without value encountered: %v", dpr)
+			continue
```

Comment:
@darkonie I had this same concern but if you look at the context of the code, partial results are returned for most if not all errors (outside of the context of this changeset)